### PR TITLE
Giphy integration

### PR DESCRIPTION
**Pull request by kngako**
Added Giphy Integration using the Giphy Android SDK and sending the giphy message in a similar fashion as it is sent in the ios app.

Also has some changes to read configs/API keys from local properties (with a `local.defaults.properties` so that the `BuildConfig` builds with some default values.

### CODE REVIEW

File: sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt [hunk 1]
```diff
@@ -428,6 +441,18 @@ abstract class ChatViewModel<ARGS: NavArgs>(
                         }
                     }
                 }
+            } else if (viewState is AttachmentSendViewState.PreviewGiphy) {
```
Comment (05nelsonm) [inline sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt:444]: can be folded into the previous if statement using or `||`

File: sphinx/application/data/concepts/repositories/concept-repository-message/build.gradle [hunk 2]
```diff
@@ -8,6 +8,7 @@ dependencies {
     api project(path: ':sphinx:application:common:wrappers:wrapper-message')
 
     implementation deps.kotlin.coroutinesCore
+    implementation deps.square.moshi
```
Comment (05nelsonm) [inline sphinx/application/data/concepts/repositories/concept-repository-message/build.gradle:11]: can be deleted

**Review by 05nelsonm — CHANGES_REQUESTED** [review]
Awesome! a couple of minor things, but looks & works great!

File: sphinx/application/data/concepts/repositories/concept-repository-message/build.gradle (hunk 2 above)
Comment (kngako) [reply sphinx/application/data/concepts/repositories/concept-repository-message/build.gradle:11]: Thanks for the catch

File: sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt (hunk 1 above)
Comment (kngako) [reply sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt:444]: There is a slight differences introduced by line 436 where we don't need to check if the file being previewed is the same one as the new one being previewed. Couldn't come up with a way to use the `||` without introducing complexity on the block of code around line 436.

**Review by 05nelsonm — APPROVED** [review]

**Pull request merged**